assets/src/components/mapPage.tsx

```diff
@@ -247,9 +242,8 @@ const MapDisplay = ({
 
   const stations = useStations()
 
-  const { socket } = useContext(SocketContext)
-
-  const selectedVehicleRef = useVehicleForIdCached(socket, selectedVehicleId),
+  const _selectedVehicleRef = useMostRecentVehicleById(selectedVehicleId),
+    selectedVehicleRef = _selectedVehicleRef && isVehicle(_selectedVehicleRef) && _selectedVehicleRef || null,
```

Comment:
**comment (non-blocking):** I realize the intent of splitting this into two separate assignments is to avoid calling hooks conditionally, but usually we use the convention from Elixir (and maybe other languages?) where a variable name prefixed with an underscore isn't used later on, so this feels _slightly_ at odds with that convention, although its one use is in the came `const` statement... so I'm conflicted about how I feel about it.

Reply:
maybe we can setup a lint rule so that we get warned when variables with a leading `_` are used?